opencti-platform/opencti-front/src/private/components/analysis/malware_analyses/MalwareAnalysisEditionOverview.tsx

```diff
@@ -0,0 +1,401 @@
+  x_opencti_workflow_id: Option
+  objectMarking?: Option[]
+}
+
+// eslint-disable-next-line max-len
+const MalwareAnalysisEditionOverview: FunctionComponent<MalwareAnalysisEditionOverviewProps> = ({ malwareAnalysisRef, context, enableReferences = false, handleClose }) => {
+  const { t } = useFormatter();
+  const basicShape = {
+    result_name: Yup.string().required(t('This field is required')),
+    product: Yup.string().required(t('This field is required')),
+    submitted: Yup.date().typeError(t('The value must be a datetime (yyyy-MM-dd hh:mm (a|p)m)')),
+    confidence: Yup.number(),
+    result: Yup.string().nullable(),
+    x_opencti_workflow_id: Yup.object(),
+  };
+  const malwareAnalysisValidator = useSchemaCreationValidation('Malware-Analysis', basicShape);
+
+  const malwareAnalysis = useFragment(malwareAnalysisEditionOverviewFragment, malwareAnalysisRef);
+
+  const createdBy = convertCreatedBy(malwareAnalysis);
+  const objectMarking = convertMarkings(malwareAnalysis);
+  const status = convertStatus(t, malwareAnalysis);
+
+  const [commitRelationAdd] = useMutation<MalwareAnalysisEditionOverviewRelationAddMutation>(malwareAnalysisMutationRelationAdd);
+  const [commitRelationDelete] = useMutation<MalwareAnalysisEditionOverviewRelationDeleteMutation>(malwareAnalysisMutationRelationDelete);
+  const [commitFieldPatch] = useMutation<MalwareAnalysisEditionOverviewFieldPatchMutation>(malwareAnalysisMutationFieldPatch);
+  const [commitEditionFocus] = useMutation<MalwareAnalysisEditionOverviewFocusMutation>(malwareAnalysisEditionOverviewFocus);
+
+  const handleChangeCreatedBy = (_: string, value: Option) => {
+    if (!enableReferences) {
+      commitFieldPatch({
+        variables: {
+          id: malwareAnalysis.id,
+          input: [{ key: 'createdBy', value: [value.value] }],
+        },
+      });
+    }
+  };
+  const handleChangeObjectMarking = (_: string, values: Option[]) => {
+    if (!enableReferences) {
+      const currentMarkingDefinitions = (malwareAnalysis.objectMarking?.edges ?? []).map(({ node }) => ({ label: node.definition, value: node.id }));
+      const added = R.difference(values, currentMarkingDefinitions).at(0);
+      const removed = R.difference(currentMarkingDefinitions, values).at(0);
+      if (added) {
+        commitRelationAdd({
+          variables: {
+            id: malwareAnalysis.id,
+            input: {
+              toId: added.value,
+              relationship_type: 'object-marking',
+            },
+          },
+        });
+      }
+      if (removed) {
+        commitRelationDelete({
+          variables: {
+            id: malwareAnalysis.id,
+            toId: removed.value,
+            relationship_type: 'object-marking',
+          },
+        });
+      }
+    }
+  };
+
+  const handleSubmitField = (name: string, value: Option | string | string[] | number | number[] | null) => {
+    if (!enableReferences) {
+      let finalValue: string = value as string;
```

Comment:
😱
So you have a value, that could be (: Option | string | string[] | number | number[] | null) but in fact it is only a string ?

Reply:
It can be also a number or array of numbers (confidence), an Option (name === 'x_opencti_workflow_id), an array of strings (openVocab)